### DISCUSSION TITLE
JK BMS and driver blocking charge after installation

### DISCUSSION THREAD

**Discussion by rodgbadger** [Louisvdw/dbus-serialbattery#641] [Q&A]
Hey folks, 

First of all thanks for all the hard work in providing such a useful driver! 

I've got two 14s battery banks with JK-BD6A17S8P BMS in parallel, connected to a Victron Multiplus. I have been trying to set up the driver to get Venus OS to communicate with the BMS',  I've got both of the battery banks showing up in the device list of Venus and discharging is working fine but unfortunately not having luck with charging.  For some reason in the IO menu allow charge is set to 'no', parameters menu lists discharge current as '0.00' and details menus lists number of modules blocking charge/discharge '10'. I have tried both the latest nightly and the current stable release but I have the same issue in both. I have changed set the cell voltage limits and discharge/charge currents in both config ini and utils.py according to release. Battery SOC is showing 75%. I have also confirmed charge/discharge is enabled via the bluetooth bms configuration. I'm a bit stuck now so any help appreciated. Am I missing something obvious?

**Comment by dfch** [comment]
Hi @rodgbadger 

 Hard to tell without seeing any logs, screenshots or configuration. 
* You mentioned you have *two* 14s battery banks installed. Are both battery banks prohibiting charging? 
* Does the system show the same behaviour when only one or the other battery is connected? 
* Are you using a battery aggregator or to which battery bank does the shown SoC refer? Does it make a difference when selecting either one or the other battery as the main battery monitor? 
* You write the *Discharge* current is `0.00`. Is this the `DCL` value? What does the `CCL` value show? And what does the driver show in the *Details* page? Normally it tells you what constraint kicks in (such as TEMP, SoC, ...).
